### ISSUE TITLE
[Bug]: MERGE fails with an error when used with hypertables 

### ISSUE THREAD

**Issue by iroussos** [timescale/timescaledb#4929]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Command processing

### What happened?

We have found that the new [MERGE command](https://www.postgresql.org/docs/current/sql-merge.html) (introduced in PG 15) fails with an error when used with hypertables. 

Adding the notes by @lkshminarayanan from the internal slack thread and we can update with further details:

I have two hypertables - conditions, conditions_updated with similar definition and overlapping data.

The following MERGE command should

1. Insert the rows from conditions_updated table into conditions table if the (time, location) key doesn't exist already in conditions
2. If the key already exists in conditions, then the conditions.temperature value is set to the average of the temperature from the two rows' temperature values.

```
MERGE INTO conditions c
USING conditions_updated cu
ON c.time = cu.time AND c.location = cu.location
WHEN MATCHED THEN
UPDATE SET temperature = (c.temperature + cu.temperature)/2
WHEN NOT MATCHED THEN
INSERT (time, location, temperature) VALUES (cu.time, cu.location, cu.temperature);
```

And I get the following error when I run this query :

```
ERROR:  invalid INSERT on the root table of hypertable "conditions"
HINT:  Make sure the TimescaleDB extension has been preloaded.
```

But the TimescaleDB extension is already installed :

```
SELECT extname  FROM pg_extension;
   extname
-------------
 plpgsql
 timescaledb
```

Exact same set of queries succeed if I don't create hypertables (i.e. skip the create_hypertable() calls).

### TimescaleDB version affected

2.8.1

### PostgreSQL version used

15.0

### What operating system did you use?

Not Applicable

### What installation method did you use?

Not applicable

### What platform did you run on?

Not applicable

### Relevant log output and stack trace

```bash
ERROR:  invalid INSERT on the root table of hypertable "conditions"
HINT:  Make sure the TimescaleDB extension has been preloaded.
```

### How can we reproduce the bug?

```bash
-- Create conditions table with location and temperature
DROP TABLE IF EXISTS conditions;
CREATE TABLE conditions (
   time        TIMESTAMPTZ       NOT NULL,
   location    SMALLINT          NOT NULL,
   temperature DOUBLE PRECISION  NULL
);
​
SELECT create_hypertable(
  'conditions',
  'time',
  chunk_time_interval => INTERVAL '5 seconds');
  
​
INSERT INTO conditions
SELECT time, location, ROUND((random()*100)) as temperature 
FROM generate_series(
	'2021-01-01 00:00:00',
    '2021-01-01 00:00:09',
    INTERVAL '5 seconds'
  ) as time, 
generate_series(1,4) as location;
​
-- Create conditions_updated table with location and temperature
DROP TABLE IF EXISTS conditions_updated;
CREATE TABLE conditions_updated (
   time        TIMESTAMPTZ       NOT NULL,
   location    SMALLINT          NOT NULL,
   temperature DOUBLE PRECISION  NULL
);
​
SELECT create_hypertable(
  'conditions_updated',
  'time',
  chunk_time_interval => INTERVAL '5 seconds');
​
-- Generate data that overlaps with conditions table
INSERT INTO conditions_updated
SELECT time, location, ROUND((random()*100)) as temperature 
FROM generate_series(
	'2021-01-01 00:00:05',
    '2021-01-01 00:00:14',
    INTERVAL '5 seconds'
  ) as time, 
generate_series(1,4) as location;
​
-- Print table/rows/num of chunks
select * from conditions order by time, location asc;
select * from conditions_updated order by time, location asc;
select hypertable_name, count(*) as num_of_chunks from timescaledb_information.chunks group by hypertable_name;
​
-- Print expected values in the conditions table once conditions_updated is merged into it
-- If a key exists in both tables, we take average of the temperature measured
-- average logic here is a mess but it works
SELECT COALESCE(c.time, cu.time) as time,
       COALESCE(c.location, cu.location) as location,
       (COALESCE(c.temperature, cu.temperature) + COALESCE(cu.temperature, c.temperature))/2 as temperature 
FROM conditions AS c FULL JOIN conditions_updated AS cu
ON c.time = cu.time AND c.location = cu.location;
​
-- Merge conditions_updated into conditions
MERGE INTO conditions c
USING conditions_updated cu
ON c.time = cu.time AND c.location = cu.location
WHEN MATCHED THEN
UPDATE SET temperature = (c.temperature + cu.temperature)/2
WHEN NOT MATCHED THEN
INSERT (time, location, temperature) VALUES (cu.time, cu.location, cu.temperature);
​
SELECT * FROM conditions ORDER BY time, location ASC;
```

**Comment by iroussos** [comment]
@horzsolt @lkshminarayanan I have added a bug report issue to follow on the bug that @lkshminarayanan has added.

We have agreed with @horzsolt that this is **not** a priority right now, the plan is to:

1. Add explicit error when MERGE is used with hypertables (follow-up issue that I am going to add)
2. Address the high priority pg15 bugs - https://github.com/timescale/timescaledb/labels/pg15
3. Release support for PG15 without support for hypertables in MERGE 
4. Asses complexity of addressing this bug - work on it in Y2023Q1 if the effort is high or after the high priority  initiatives for Y2022Q4 are completed 

In order for us to be able to plan for this properly, I have added this on the backlog of the bug rotation board and on the candidates section of our product roadmap board

FYI @jfjoly following our discussion earlier today
